frontend/test/pytest/test_static_arguments.py

```diff
@@ -0,0 +1,142 @@
+# Copyright 2022-2023 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Test features related to static arguments."""
+
+from dataclasses import dataclass
+import pytest
+
+from catalyst import qjit
+
+class TestStaticArguments:
+    """Test QJIT with static arguments."""
+
+    def test_zero_static_argument(self):
+        """Test QJIT with zero static argument."""
+
+        @qjit
+        def f0(
+            x: int,
+        ):
+            return x
+
+        @qjit(static_argnums=())
+        def f1(
+            x: int,
+        ):
+            return x
+
+        @qjit(static_argnums=None)
+        def f2(
+            x: int,
+        ):
+            return x
+
+        assert f0(1) == 1
+        assert f1(2) == 2
+        assert f2(3) == 3
+
+    def test_out_of_bounds_static_argument(self):
+        """Test QJIT with invalid static argument index."""
```

Comment:
Should this test case raise an error?

Reply:
I made the compiler ignore the invalid values. I will change that to raise an error.